addons/mail/static/tests/new/thread/thread_tests.js

```diff
@@ -586,3 +586,45 @@ QUnit.test("Mention a partner with special character (e.g. apostrophe ')", async
         `.o_mail_redirect[data-oe-id="${resPartnerId1}"][data-oe-model="res.partner"]:contains("@Pynya's spokesman")`
     );
 });
+
+QUnit.test("mention 2 different partners that have the same name", async function (assert) {
+    const pyEnv = await startServer();
+    const [resPartnerId1, resPartnerId2] = pyEnv["res.partner"].create([
+        {
+            email: "partner1@example.com",
+            name: "TestPartner",
+        },
+        {
+            email: "partner2@example.com",
+            name: "TestPartner",
+        },
+    ]);
+    const mailChannelId1 = pyEnv["mail.channel"].create({
+        name: "test",
+        channel_member_ids: [
+            [0, 0, { partner_id: pyEnv.currentPartnerId }],
+            [0, 0, { partner_id: resPartnerId1 }],
+            [0, 0, { partner_id: resPartnerId2 }],
+        ],
+    });
+    const { openDiscuss } = await start();
+    await openDiscuss(mailChannelId1);
+    await insertText(".o-mail-composer-textarea", "@");
+    await insertText(".o-mail-composer-textarea", "T");
+    await insertText(".o-mail-composer-textarea", "e");
+    await afterNextRender(() => target.querySelectorAll(".o-composer-suggestion")[0].click());
```

Comment:
Not sure if we care, I think we don't but could have use `querySelector` instead of `querySelectorAll(...)[0]`

Reply:
It feel a bit silly to use querySelector just because it default to the [0] key.